ObjectiveZipLib/Objective-Zip/UnzipWithProgress.mm

```diff
@@ -182,8 +182,9 @@ - (void) unzipToURL:(NSURL *)destinationFolder
       FileInZipInfo * info = [_zipTool getCurrentFileInZipInfo];
       
       NSError * error  = nil;
-      if ( !self.unzipFileDelegate || [self.unzipFileDelegate includeFileWithName:info.name error:&error] )
+      if ( !self.unzipFileDelegate || [self.unzipFileDelegate includeFileWithName:info.name forDestinationURL:destinationFolder error:&error] )
       {
+         [_zipTool locateFileInZip:info.name];
```

Comment:
Could we do this before the `if ... ` since both clauses do it?

Reply:
No, because the `if ...` is what changes the file pointer in `_zipTool`.  This is not ideal.  I would prefer the delegate method be responsible for restoring the state, but that gets messy. :(

Reply:
Ah. I wondered if that might be the case.